### ISSUE TITLE
Regression in test_major_compaction_processes_tables_in_order_by_size dtests

### ISSUE THREAD

**Issue by bhalevy** [scylladb/scylladb#12811]
Seen e.g. in https://jenkins.scylladb.com/view/master/job/scylla-master/job/dtest-daily-release/192/testReport/compaction_additional_test/TestCompactionAdditional/Run_Dtest_Parallel_Cloud_Machines___FullDtest___full_split000___test_major_compaction_processes_tables_in_order_by_size_100_10_000_100_000_/
```
AssertionError: The list of rows sorted by size is not identical to the list of rows sorted by compaction time
assert [CfSizeTime(name='cf_3',\n            size=100000,\n            compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 742000)),\n CfSizeTime(name='cf_2',\n            size=10000,\n            compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 781000)),\n CfSizeTime(name='cf_1',\n            size=100,\n            compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 788000))] == [CfSizeTime(name='cf_1',\n            size=100,\n            compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 788000)),\n CfSizeTime(name='cf_2',\n            size=10000,\n            compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 781000)),\n CfSizeTime(name='cf_3',\n            size=100000,\n            compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 742000))]
  At index 0 diff: TestCompactionAdditional.test_major_compaction_processes_tables_in_order_by_size.<locals>.CfSizeTime(name='cf_3', size=100000, compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 742000)) != TestCompactionAdditional.test_major_compaction_processes_tables_in_order_by_size.<locals>.CfSizeTime(name='cf_1', size=100, compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 788000))
  Full diff:
    [
  +  CfSizeTime(name='cf_3',
  +             size=100000,
  +             compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 742000)),
  +  CfSizeTime(name='cf_2',
  +             size=10000,
  +             compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 781000)),
     CfSizeTime(name='cf_1',
                size=100,
                compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 788000)),
  -  CfSizeTime(name='cf_2',
  -             size=10000,
  -             compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 781000)),
  -  CfSizeTime(name='cf_3',
  -             size=100000,
  -             compaction_time=datetime.datetime(2023, 2, 11, 6, 45, 36, 742000)),
    ]
```

It looks like this has the same cause as #12808: https://github.com/scylladb/scylladb/commit/55a8421e3d35b85ca1497d615540aafe2e02bffd
confusing `force_keyspace_compaction` sorting as the table stats aren't updated and I guess their data size is 0.

Cc @raphaelsc

**Comment by bhalevy** [comment]
@mykaul since the offending patch is confined to master, do we still want to mark the issue as regression?

**Comment by bhalevy** [comment]
should be fixed by d6fe99abc452331ff3ad44ac8553f7fe076cb11a
